nginx/nginx.tmpl

```diff
@@ -102,9 +104,21 @@ http {
 
 {{- range $entry := .Servers }}
     # ingress: {{ $entry.Name }}
+  {{- range $IngressPortConf := $IngressPorts }}
     server {
-        listen {{ $port }}{{ if $proxyprotocol }} proxy_protocol{{ end }};
+        listen {{ $IngressPortConf.Port }}{{ if $proxyprotocol }} proxy_protocol{{ end }}{{- if eq $IngressPortConf.Name "https" }} ssl {{ end }};
         server_name {{ $entry.ServerName }};
+{{- if eq $IngressPortConf.Name "https" }}
+        ssl_certificate     {{ $DefaultSslPath }}.crt;
+        ssl_certificate_key  {{ $DefaultSslPath }}.key;
+        # secure settings (A+ at SSL Labs ssltest at time of writing)
+        # see https://wiki.mozilla.org/Security/Server_Side_TLS#Nginx
```

Comment:
Please link the exact profile you used so we can update it in the future easily.

Reply:
https://mozilla.github.io/server-side-tls/ssl-config-generator/ - Nginx, Modern Profile is what we should use. Possibly adding TLSv1, TLSv1.1 as additional SSL protocols.

Reply:
If possible we could eliminate TLSv1 and TLS v1.1. We only really need to support those if we're going to allow direct connections from old client devices. Everything via Akamai or JVM 8 should be fine with TLS v1.2.